enterprise/backend/test/metabase_enterprise/advanced_permissions/common_test.clj

```diff
@@ -169,22 +169,112 @@
+                    Table    [t3 {:db_id db-id, :schema "schema1", :name "t3"}]]
       (with-all-users-data-perms {db-id {:data       {:schemas :all :native :write}
-                                           :data-model {:schemas :all}}}
-        (testing "Testing /:id/schema/:schema variants"
-          (perms/revoke-data-perms! (perms-group/all-users) db-id)
-          (testing "/:id/schema/:schema when permissions are revoked but include_editable_data_model=true returns values"
-            (is (= ["t1" "t3"]
-                   (map :name (mt/user-http-request :rasta :get 200 (format "database/%d/schema/%s" db-id "schema1")
-                                                    :include_editable_data_model true)))))
-          (testing "/:id/schema/:schema when permissions are revoked ins a 403"
-            (is (= "You don't have permissions to do that."
-                   (mt/user-http-request :rasta :get 403 (format "database/%d/schema/%s" db-id "schema1"))))))))))
+                                         :data-model {:schemas :all}}}
+        (perms/revoke-data-perms! (perms-group/all-users) db-id)
+        (testing "If data permissions are revoked, it should be a 403"
+          (is (= "You don't have permissions to do that."
+                 (mt/user-http-request :rasta :get 403 (format "database/%d/schema/%s" db-id "schema1")))))
+        (testing "If include_editable_data_model=true and data permissions are revoked, it should return values"
+          (is (= ["t1" "t3"]
+                 (map :name (mt/user-http-request :rasta :get 200 (format "database/%d/schema/%s" db-id "schema1")
+                                                  :include_editable_data_model true))))))
+
+      (testing "If include_editable_data_model=true and a non-admin does not have data model perms, it should respond
+                with a 404"
+        (with-all-users-data-perms {db-id {:data       {:schemas :all :native :write}
+                                           :data-model {:schemas :none}}}
+          (is (= "Not found."
+                 (mt/user-http-request :rasta :get 404 (format "database/%d/schema/%s" db-id "schema1")
+                                       :include_editable_data_model true)))))
+
+      (testing "If include_editable_data_model=true and a non-admin has data model perms for a single table in a schema,
+                the table is returned"
+        (with-all-users-data-perms {db-id {:data       {:schemas :all :native :write}
+                                           :data-model {:schemas {"schema1" {(u/the-id t1) :all
+                                                                             (u/the-id t3) :none}}}}}
+          (is (= ["t1"]
+                 (map :name (mt/user-http-request :rasta :get 200 (format "database/%d/schema/%s" db-id "schema1")
+                                                  :include_editable_data_model true)))))))))
+
+(deftest get-schema-with-empty-name-and-advanced-perms-test
```

Comment:
Permissions with the `GET "api/database/:id/schema/"` endpoint were untested, so I added them. This is essentially the same set of tests as the ``GET "api/database/:id/schema/:id"` endpoint, with some slight changes.